sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -103,6 +106,13 @@ object DataType {
 
   def fromJson(json: String): DataType = parseDataType(parse(json))
 
+  // Until Spark-2.1, we use json strings for defining schemas in user-facing APIs.
+  // Since we add an user-friendly API in the DDL parser, we employ DDL formats for the case.
+  // To keep back-compatibility, we use `fromJson` first, and then try the new API.
+  def fromString(text: String): DataType = {
```

Comment:
In this pr, I just propose we support new DDL formats in the existing APIs that's already used json formats for defining schemas (e.g., `functions.from_json`). So, we need to support both json formats and DDL formats there. If I misunderstood you, could you correct me? Thanks for your comment!